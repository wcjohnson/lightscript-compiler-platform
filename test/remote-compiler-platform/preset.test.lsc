import { InPlacePlatform } from '../..'

jest.setTimeout(100000000)

test! "preset toolchain", -/>
  p = new InPlacePlatform()
  debugger
  t <- p.createTool({
    source: { type: 'toolchain' }
    options: {
      children: [
        // babel
        {
          source: {
            type: 'remote'
            url: 'https://wcjohnson.github.io/lightscript-compiler/modules/platform/babel-7.1.0.min.js'
          }
        }
        // lsc preset
        {
          source: {
            type: 'remote'
            format: 'cjs'
            url: 'https://wcjohnson.github.io/lightscript-compiler/modules/preset/4.0.0-alpha.19.min.js'
          }
        }
      ]
    }
  })
  description <- t.describe(true)
  act <- t.apply(null)
  rst <- act.run({
    code: `
      f() -> 1
    `
  })
  console.log("Preset result", rst)
  expect({ description, result: rst}).toMatchSnapshot()
