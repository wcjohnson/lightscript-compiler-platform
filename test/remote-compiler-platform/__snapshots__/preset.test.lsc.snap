// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preset toolchain 1`] = `
Object {
  "description": Object {
    "_metadata": Object {
      "hasChildren": true,
      "name": "Toolchain",
      "possibleOptions": Object {
        "children": Object {
          "mutable": true,
          "valueType": "array(tool)",
        },
      },
    },
    "options": Object {
      "children": Array [
        Object {
          "_metadata": Object {
            "hasChildren": false,
            "name": "Babel Compiler Platform: v7.0.0",
            "possibleOptions": Object {},
            "types": Object {
              "input": Object {
                "name": "NullAction",
              },
              "output": Object {
                "name": "BabelCompileAction",
              },
            },
          },
          "source": Object {
            "type": "remote",
            "url": "https://wcjohnson.github.io/lightscript-compiler/modules/platform/babel-7.1.0.min.js",
          },
        },
        Object {
          "_metadata": Object {
            "hasChildren": false,
            "name": "LightScript Preset: v4.0.0-alpha.19",
            "possibleOptions": Object {
              "options": Object {
                "defaultValue": "{}",
                "description": "LightScript preset options (JSON)",
                "valueType": "string",
              },
            },
            "types": Object {
              "input": Object {
                "name": "BabelCompileAction",
              },
              "output": Object {
                "name": "BabelCompileAction",
              },
            },
            "ui": Object {
              "noLiveUpdates": true,
            },
          },
          "options": Object {
            "options": undefined,
          },
          "source": Object {
            "format": "cjs",
            "type": "remote",
            "url": "https://wcjohnson.github.io/lightscript-compiler/modules/preset/4.0.0-alpha.19.min.js",
          },
        },
      ],
    },
    "source": Object {
      "type": "toolchain",
    },
  },
  "result": Object {
    "ast": null,
    "code": "\\"use strict\\";

function f() {
  return 1;
}",
    "map": null,
  },
}
`;
